src/android/com/microsoftopentechnologies/intellij/helpers/DatePickerCellEditor.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2014 Microsoft Open Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.microsoftopentechnologies.intellij.helpers;
+
+import com.intellij.ui.JBColor;
+import org.jdesktop.swingx.JXMonthView;
+import org.jdesktop.swingx.calendar.DateSelectionModel;
+import sun.reflect.misc.ReflectUtil;
+import sun.swing.SwingUtilities2;
+
+import javax.swing.*;
+import javax.swing.border.LineBorder;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.lang.reflect.Constructor;
+import java.text.SimpleDateFormat;
+
+public abstract class DatePickerCellEditor extends DefaultCellEditor {
+
+    Constructor constructor;
+    Object value;
+
+    public DatePickerCellEditor() {
+        super(new JTextField());
+        this.getComponent().setName("Table.editor");
+    }
+
+    @Override
+    public boolean stopCellEditing() {
+        String var1 = (String)super.getCellEditorValue();
+
+        try {
+            if("".equals(var1)) {
+                if(this.constructor.getDeclaringClass() == String.class) {
+                    this.value = var1;
+                }
+
+                super.stopCellEditing();
+            }
+
+            SwingUtilities2.checkAccess(this.constructor.getModifiers());
```

Comment:
Based in the testing I made while developing, the call to the checkAccess is needed and I didn't find a way to avoid it yet. I would leave it as it is until we find a workaround.